src/Interpreters/ZooKeeperLog.h

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include <Core/NamesAndAliases.h>
+#include <Interpreters/SystemLog.h>
+#include <Interpreters/ClientInfo.h>
+#include <Common/ZooKeeper/IKeeper.h>
+
+
+namespace DB
+{
+
+struct ZooKeeperLogElement
+{
+    enum Type
+    {
+        UNKNOWN = 0,
+        SEND = 1,
+        RECEIVE = 2,
+        FINALIZE = 3
+    };
+
+    Type type = UNKNOWN;
+    Decimal64 event_time = 0;
+    Poco::Net::SocketAddress address;
+    Int64 session_id = 0;
+
+    /// Common request info
+    Int32 xid = 0;
+    bool has_watch = false;
+    Int32 op_num = 0;
+    String path;
+
+    /// create, set
+    String data;
+
+    /// create
+    bool is_ephemeral = false;
+    bool is_sequential = false;
+
+    /// remove, check, set
+    std::optional<Int32> version = 0;
```

Comment:
So this optional always has value? Why it's optional?

Reply:
It should not have value if request does not support version check, I forgot to remove assignment